x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java

```diff
@@ -144,6 +159,36 @@ public Map<String, Map<String, Column>> tables() {
         return tables;
     }
 
+    /**
+     * Get a {@link LocalRelation} from the {@link #tables()} or {@code null}
+     * if there's no parameter with that name. This will cache the result so
+     * the same name will always return the same {@linkplain LocalRelation}.
+     * That isn't strictly needed, but it feels nice.
+     */
+    public LocalRelation localRelationFromTable(String name) {
+        LocalRelation localRelation = tablesFromLocalRelation.get(name);
+        if (localRelation != null) {
+            return localRelation;
+        }
+        Map<String, Column> table = tables.get(name);
+        if (table == null) {
+            return null;
+        }
+        Block[] blocks = new Block[table.size()];
+
+        List<Attribute> attributes = new ArrayList<>(blocks.length);
+        int i = 0;
+        for (Map.Entry<String, Column> entry : table.entrySet()) {
+            Column column = entry.getValue();
+            EsField field = new EsField(entry.getKey(), column.type(), Map.of(), false, false);
+            attributes.add(new FieldAttribute(Source.EMPTY, null, entry.getKey(), field));
+            blocks[i++] = column.values();
+        }
+        localRelation = new LocalRelation(Source.EMPTY, attributes, LocalSupplier.of(blocks));
```

Comment:
I _think_ we're about to lay the ground for subtle bugs for multiple lookups. When there's multiple LOOKUPs in a query that use the same table, the second should replace the attributes added by the first; but they're gonna be the exact same attributes and thus be indistinguishable from each other!

Your test illustrates this well:
```
              FROM test
            | RENAME languages AS int
            | LOOKUP int_number_names ON int
// now we have `name` and it spells out the number of languages
            | RENAME name AS languages_name, int AS languages
            | EVAL int = LENGTH(last_name)
            | LOOKUP int_number_names ON int
// now `name` should spell out the length of the last name
```
After each LOOKUP, the `name` attribute is exactly the same, down to the attribute id, even though shadowing did happen. On the level of attributes, the second LOOKUP was a noop.

The attribute id is used to resolve shadowing situations, so this is not good.

(For reference: This situation can also come up in SQL implementations, but for JOINs SQL normally requires to have distinct table names/aliases which then become qualifiers of the attribute name; in this sense, SQL also has distinct attributes even when joining multiple times with the same table.)

Reply:
It *sounds* like this is a good argument for not doing this change at all.

Reply:
> I think I'd prefer not doing this change at all.

Easy. Closed.